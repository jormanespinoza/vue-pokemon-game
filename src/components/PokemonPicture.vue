<template>
  <div class="pokemon-container">
    <img :src="imageSrc" class="hidden-pokemon" alt="Pokemon" />
    <img v-if="showPokemon" :src="imageSrc" class="fade-in" alt="Pokemon" />
  </div>
</template>

<script>
export default {
  props: {
    pokemonId: {
      type: Number,
      required: true
    },
    showPokemon: {
      type: Boolean,
      required: true,
      default: false
    }
  },
  computed: {
    imageSrc() {
      return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
    }
  }
}
</script>

<style scope>
/* Pokemon Picture */
.pokemon-container {
  height: 200px;
}

img {
  height: 200px;
  position: absolute;
  left: 50%;
  transform: translateX(-50%);
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  -o-transform: translateX(-50%);
  user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-drag: none;
  -webkit-user-select: none;
}

.hidden-pokemon {
  filter: brightness(0);
}
</style>