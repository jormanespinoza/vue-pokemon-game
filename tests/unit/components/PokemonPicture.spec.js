import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture.vue', () => {

    test('should match snapshot', () => {
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId: 25,
                showPokemon: false
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    test('should show hidden pokemon and pokemon with 1', () => {
        const pokemonId = 1
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId,
                showPokemon: false
            }
        })

        const [hiddenPokemonImage, pokemonImage] = wrapper.findAll('img')

        expect(hiddenPokemonImage.exists()).toBeTruthy()
        expect(hiddenPokemonImage.classes('hidden-pokemon')).toBeTruthy()
        expect(pokemonImage).toBe(undefined)
        expect(hiddenPokemonImage.attributes('src'))
            .toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemonId}.svg`)
    })

    test('should show pokemon when showPokemon is true', () => {
        const pokemonId = 25
        const wrapper = shallowMount(PokemonPicture, {
            props: {
                pokemonId,
                showPokemon: true
            }
        })

        const [, pokemonImage] = wrapper.findAll('img')

        expect(pokemonImage.exists()).toBeTruthy()
        expect(pokemonImage.classes('fade-in')).toBeTruthy()
        expect(pokemonImage.attributes('src'))
            .toBe(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${pokemonId}.svg`)
    })
})