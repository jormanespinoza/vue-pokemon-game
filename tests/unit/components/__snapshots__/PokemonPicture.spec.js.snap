// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture.vue should match snapshot 1`] = `
<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/25.svg" class="hidden-pokemon" alt="Pokemon">
  <!--v-if-->
</div>
`;
